### PR TITLE
Use g_return_val_if_fail() to fix build on OS X

### PR DESCRIPTION
**Pull request by techee**

### CODE REVIEW

File: src/plugins.c [hunk 1]
```diff
@@ -2000,7 +2000,7 @@ gboolean geany_plugin_register_proxy(GeanyPlugin *plugin, const gchar **extensio
 	foreach_list(node, active_proxies.head)
 	{
 		proxy = node->data;
-		g_return_if_fail(p != proxy->plugin);
+		g_return_val_if_fail(p != proxy->plugin, FALSE);
```
Comment (elextr) [inline src/plugins.c]: @kugel- based on the comment above the loop, why does this need to print an error if the plugin is already registered?

The `g_return_*` functions are only meant for programming errors, see [here](https://developer.gnome.org/glib/stable/glib-Warnings-and-Assertions.html#glib-Warnings-and-Assertions.description)

Comment (kugel-) [inline src/plugins.c]: Registering twice is currently explicitly unsupported currently, so any attempt is a programming error (though a plugin error so an assert might not be ideal)

The plan is to enable it once the details are fleshed out and code is written, with the intention to enable proxies to change their list of supported files

Comment (elextr) [inline src/plugins.c]: ok, thanks, carry on :)

Comment (kugel-) [inline src/plugins.c]: The diff doesn't show but there's a comment about that just above ;-)

/me is sitting on the airport at just after midnight waiting for the flight

**Comment by b4n** [comment]
LGTM.  Damn this GCC change hiding those missing return values in `g_return*_if_fail()` :S

**Comment by codebrainz** [comment]
It would probably be better to unconditionally check and then print a useful error than to use one of these debug assertions, which will, if enabled, only propagate the error, making it harder to find. I guess that's kind of another issue than this PR aims to solve though.

**Comment by b4n** [comment]
> LGTM.  Damn this GCC change hiding those missing return values in `g_return*_if_fail()` :S

https://bugzilla.gnome.org/show_bug.cgi?id=753310 yay!

**Pull request merged**